### PR TITLE
HPCC-19056 Update build instructions

### PR DESCRIPTION
**Pull request by shamser**
Signed-off-by: Shamser Ahmed <shamser.ahmed@lexisnexis.co.uk>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by shamser** [comment]
@richardkchapman @ghalliday @jakesmith I have used the build instructions from hpccsystems github wiki to update the BUILD_ME.md.  The content is nearly identical.  I have reformatted, removed some duplicates and re-organized a little.

File: BUILD_ME.md [hunk 1]
```diff
@@ -13,95 +13,305 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
-sudo apt-get install libboost-all-dev
-sudo apt-get install libssl-dev
-sudo apt-get install libapr1-dev
-sudo apt-get install libaprutil1-dev
-sudo apt-get install clang
-sudo apt-get install libtool
-sudo apt-get install autotools-dev
-sudo apt-get install automake
-```
+* Visit [https://github.com/hpcc-systems/HPCC-Platform/wiki](https://github.com/hpcc-systems/HPCC-Platform/wiki) for upto date help and guidance.
```
Comment (jakesmith) [inline BUILD_ME.md]: minor: should be 'up to date' I think.

File: BUILD_ME.md [hunk 2]
```diff
@@ -13,95 +13,305 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
+
+```sudo yum install -y epel-release```
+
+#### CentOS 6.4:
+
+    sudo yum install gcc-c++ gcc make bison flex binutils-devel openldap-devel libicu-devel libxslt-devel libarchive-devel boost-devel openssl-devel \
+    apr-devel apr-util-devel hiredis-devel numactl-devel libmysqlclient-dev libevent-devel tbb-devel atlas-devel python34 R-core-devel R-Rcpp-devel \
+    R-inline R-RInside-devel nodejs libcurl-devel
+
+#### CentOS 7:
```
Comment (jakesmith) [inline BUILD_ME.md]: minor: not new (same as wiki) but Ubuntu (rightly I think) the latest version 1st, so probably this should list CentOS 7 before CentOS 6.4 here too (and in Wiki).

**Comment by jakesmith** [comment]
2 minor comments, but looks good.
It's a pity there isn't 1 source for this.
Could the Wiki point to a nicely format version of this in our source tree instead.. i.e. this was primary

**Comment by jakesmith** [comment]
@richardkchapman @ghalliday - can you look/comment too please.

**Comment by ghalliday** [comment]
the checkboxes should be filled in.

File: BUILD_ME.md (hunk 1 above)
Comment (shamser) [reply BUILD_ME.md]: I'm not sure why this confused me so much.  I've learnt something new here: https://www.lexico.com/definition/up_to_date.  Apparently, compound adjectives should be hyphenated if used before the noun but not if it's used after!

**Review by ghalliday — COMMENTED** [review]
@shamser sorry, I hadn't committed review changes

File: BUILD_ME.md [hunk 3]
```diff
@@ -13,95 +13,305 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
...
+git clone https://github.com/Microsoft/vcpkg
+cd vcpkg
+bootstrap-vcpkg
 ```
-* To build the makefiles just created above, run
+
+Use vcpkg to install various packages:
 ```
-   make
+vcpkg install zlib
```
[the remark this answers is not in the document]
Comment (shamser) [reply BUILD_ME.md]: @ghalliday I haven't verified any of the Windows related build instructions.  Shall I remove all the Windows instructions for now?

[the remark this answers is not in the document]
Comment (ghalliday) [reply BUILD_ME.md]: I think leave them.  Someone else tried them and seemed to make some progress.

**Comment by ghalliday** [comment]
@richardkchapman looks fine to me.

**Comment by richardkchapman** [comment]
@shamser Please squash

**Comment by shamser** [comment]
@richardkchapman Squashed

**Pull request merged**